test-skripsi/test-improvement.tsx

```diff
@@ -23,11 +23,25 @@
 	}
 
 	getAllData(): Array<T> {
-		return this.data;
+		return [...this.data];
 	}
 
 	sortData(compareFunction: (a: T, b: T) => number): void {
-		this.data.sort(compareFunction);
+		if (this.data.length > 1000) {
+			this.data.sort(compareFunction);
+		} else {
+			this.insertionSort(compareFunction);
+		}
+	}
+
+	private insertionSort(compareFunction: (a: T, b: T) => number): void {
+		for (let i = 1; i < this.data.length; i++) {
+			let j = i;
+			while (j > 0 && compareFunction(this.data[j - 1], this.data[j]) > 0) {
+				[this.data[j - 1], this.data[j]] = [this.data[j], this.data[j - 1]];
+				j--;
+			}
+		}
```

Comment:
The conditional logic for choosing the sorting algorithm based on the dataset size is a thoughtful optimization. However, it's important to document the rationale behind the threshold of 1000 items and the choice of insertion sort for smaller datasets. Including benchmarks or references can provide insight into these decisions.